### PR TITLE
Stub the web applet

### PR DESCRIPTION
**Pull request by bylaws**
Used by SMO, SM3DAS etc

### CODE REVIEW

File: app/src/main/cpp/skyline/applet/offline_web_applet.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+#pragma once
+
+#include <services/am/applet/IApplet.h>
+
+namespace skyline::applet {
+    /**
+     * @brief The Offline Web is utilized by the guest to display offline web pages
+     */
+    class OfflineWebApplet : public service::am::IApplet, service::am::EnableNormalQueue {
```
Comment (Morph1984) [inline app/src/main/cpp/skyline/applet/offline_web_applet.h:13]: Perhaps just call this WebApplet due to the ShimKind enum below

File: app/src/main/cpp/skyline/applet/offline_web_applet.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                                       std::shared_ptr<kernel::type::KEvent> onInteractiveDataPushFromApplet,
+                                       service::applet::LibraryAppletMode appletMode)
+        : IApplet{state, manager, std::move(onAppletStateChanged), std::move(onNormalDataPushFromApplet), std::move(onInteractiveDataPushFromApplet), appletMode} {}
+
+    Result OfflineWebApplet::Start() {
+        auto commonArg{PopNormalInput<service::applet::CommonArguments>()};
+        auto argHeader{PopNormalInput<WebArgHeader>()};
+
+        if ((commonArg.apiVersion >= 0x80000 && argHeader.shimKind == ShimKind::Web) || (commonArg.apiVersion >= 0x30000 && argHeader.shimKind == ShimKind::Share))
+            Logger::Error("OfflineWeb TLV output is unsupported!");
```
Comment (Morph1984) [inline app/src/main/cpp/skyline/applet/offline_web_applet.cpp:25]:
```suggestion
            Logger::Error("Web TLV output is unsupported!");
```

File: app/src/main/cpp/skyline/applet/offline_web_applet.h [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        };
+
+        /**
+         * @brief Common return value struct for all web-applet commands
+         */
+        struct WebCommonReturnValue {
+            WebExitReason exitReason;
+            u32 _pad_;
+            std::array<char, 0x1000> lastUrl;
+            std::array<char, 0x800> lastUrlSize;
```
Comment (Morph1984) [inline app/src/main/cpp/skyline/applet/offline_web_applet.h:48]:
```suggestion
            u64 lastUrlSize;
```

File: app/src/main/cpp/skyline/applet/offline_web_applet.h [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        };
+        static_assert(sizeof(WebCommonReturnValue) == 0x1808);
```
Comment (Morph1984) [inline app/src/main/cpp/skyline/applet/offline_web_applet.h:50]:
```suggestion
        static_assert(sizeof(WebCommonReturnValue) == 0x1010);
```

File: app/src/main/cpp/skyline/applet/offline_web_applet.h (hunk 3 above)
Comment (bylaws) [reply app/src/main/cpp/skyline/applet/offline_web_applet.h:48]: Ooops! thanks for catching this

File: app/src/main/cpp/skyline/applet/offline_web_applet.cpp [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+        PushNormalDataAndSignal(std::make_shared<service::am::ObjIStorage<WebCommonReturnValue>>(state, manager, WebCommonReturnValue{
+            .exitReason = WebExitReason::WindowClosed
```
[the remark this answers is not in the document]
Comment (Morph1984) [reply app/src/main/cpp/skyline/applet/offline_web_applet.cpp]: Just realized, this should be `http` not `https`

[the remark this answers is not in the document]
Comment (Morph1984) [reply app/src/main/cpp/skyline/applet/offline_web_applet.cpp]: The trailing slash should also be included

File: app/src/main/cpp/skyline/applet/web_applet.cpp [hunk 6]
```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2022 Skyline Team and Contributors (https://github.com/skyline-emu/)
+// Copyright © 2020 yuzu Emulator Project (https://github.com/yuzu-emu/)
+
+#include <input.h>
+#include <input/npad.h>
+#include <services/applet/common_arguments.h>
+#include <services/am/storage/ObjIStorage.h>
+#include "offline_web_applet.h"
```
Comment (AbandonedCart) [inline app/src/main/cpp/skyline/applet/web_applet.cpp:9]: I think this was supposed to be "web_applet.h" or the file name wasn't updated to match.

**Comment by PixelyIon** [comment]
@bylaws Why not resolve CR comments if you've presumably fixed them?

**Pull request merged**